code/game/objects/items/devices/lightreplacer.dm

```diff
@@ -95,8 +97,33 @@
 				user.drop_item()
 				qdel(L)
 				return
-		else
-			to_chat(user, "You need a working light.")
+		else if(L.status == 2 || L.status == 3)
+			AddShards(1)
+			to_chat(user, "You insert the [L.name] into the [src.name]. You have [shards] shards remaining.")

```

Comment:
to_chat(user, "You insert the [L] into the [src]. You have [shards] shards remaining.")